src/test/java/io/jenkins/plugins/services/impl/HttpClientGithubReleasesTest.java

```diff
@@ -0,0 +1,126 @@
+package io.jenkins.plugins.services.impl;
+
+import com.github.tomakehurst.wiremock.junit.WireMockRule;
+import io.jenkins.plugins.models.Plugin;
+import io.jenkins.plugins.models.PluginRelease;
+import io.jenkins.plugins.models.PluginReleases;
+import io.jenkins.plugins.models.Scm;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.time.Instant;
+import java.util.Date;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.get;
+import static com.github.tomakehurst.wiremock.client.WireMock.okJson;
+import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
+import static org.junit.Assert.assertEquals;
+
+public class HttpClientGithubReleasesTest {
+  @Rule
+  public WireMockRule wireMockRule = new WireMockRule();
+
+  private HttpClientGithubReleases httpClientGithubReleases;
+
+  static final private Plugin lighthouseReportPlugin = new Plugin() {
+    @Override
+    public Scm getScm() {
+      return new Scm() {
+        @Override
+        public String getLink() {
+          return "https://github.com/jenkinsci/lighthouse-report-plugin";
+        }
+      };
+    }
+  };
+  static final private Plugin badScmPlugin = new Plugin() {
+    @Override
+    public Scm getScm() {
+      return new Scm() {
+        @Override
+        public String getLink() {
+          return "https://randomwebsite.com/for/some/reason";
+        }
+      };
+    }
+  };
+
+  @Before
+  public void setUp() {
+    // wireMockRule.enableRecordMappings()
+
+    this.httpClientGithubReleases = new HttpClientGithubReleases(new DefaultConfigurationService() {
+      @Override
+      public String getGithubApiBase() {
+        return wireMockRule.baseUrl();
+      }
+    });
+  }
+
+  @Test
+  public void getReleasesHappy() throws Exception {
+    stubFor(get("/repos/jenkinsci/lighthouse-report-plugin/releases?client_id=null&client_secret=null")
+      .willReturn(okJson("[\n" +
+        "  {\n" +
```

Comment:
Much easier to maintain if this is extracted to a file imo, you’ll get ide highlighting and can tell where it’s broken my using your ide when you’re updating it

Reply:
I am open to a patch or pr.

Reply:
for the life of me i've never understood or gotten java resources stuff working right, feel free to suggest or change, but i don't feel like fighting it.

Reply:
normally it's:
```java
try (InputStream in = getClass.getResourceAsStream("resource.json")) {
  String resource =IOUtils.toString(in, StandardCharsets.UTF8);
}
```

and the resource file should be in the same packages as your class i.e. `src/test/resources/io/jenkins/plugins/services/impl/`

example here:
https://github.com/jenkinsci/dark-theme/pull/77/files#diff-94d9ca930eab10dbaf09d3fdc02508abR36

direct link if the diff doesn't work: https://github.com/jenkinsci/dark-theme/blob/add-plugin/src/main/java/io/jenkins/plugins/darktheme/DarkThemeManagerFactory.java#L36

Reply:
Okay used wiremock recordings instead, easier to update